lib/pdf/core/object_store.rb

```diff
@@ -19,6 +19,20 @@ def initialize(opts = {})
 
         @info  ||= ref(opts[:info] || {}).identifier
         @root  ||= ref(Type: :Catalog).identifier
+
+        if opts[:enable_pdfa_1b]
+          # PDF/A-1b requirement: XMP metadata
+          @xmp_metadata ||= ref(Type: :Metadata, Subtype: :XML).identifier
+          root.data[:Metadata] = xmp_metadata
+          xmp_metadata_content = XmpMetadata.new(opts[:info] || {})
+          xmp_metadata_content.enable_pdfa_1b = true
+          xmp_metadata.stream = Stream.new
+          xmp_metadata.stream << xmp_metadata_content.render
```

Comment:
`xmp_metadata` is a reference. It's smart enough to create a stream when needed so this can be a bit simpler:

```suggestion
          xmp_metadata << xmp_metadata_content.render
```